pax-exam-container-carbon/org.wso2.carbon.container/src/main/java/org/wso2/carbon/container/ArchiveExtractor.java

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.container;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.ops4j.pax.exam.TestContainerException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URL;
+import java.nio.file.Path;
+
+/**
+ * Extract zip or tar.gz archives to a target Directory.
+ */
+public class ArchiveExtractor {
+
+    private static final Logger logger = LoggerFactory.getLogger(ArchiveExtractor.class);
+
+    private ArchiveExtractor() {
+    }
+
+    /**
+     * Extract zip or tar.gz specified by maven url to a target Directory.
+     *
+     * @param sourceURL       url of the archive to extract
+     * @param targetDirectory where to extract to
+     * @throws IOException on I/O error
+     */
+    public static void extract(URL sourceURL, File targetDirectory) throws IOException {
+        if (sourceURL.getProtocol().equals("file")) {
```

Comment:
"file", ".zip" and several strings can be moved to constant class in this method and reuse later